src/servant/tt_004/sword.c

```diff
@@ -423,7 +423,146 @@ void func_us_80173CB8(Entity* self) {
     }
 }
 
-INCLUDE_ASM("servant/tt_004/nonmatchings/sword", ServantInit);
+extern MATRIX D_us_80178B54;
+extern s32 D_us_80178B84;
+extern s32 D_us_80178B7C;
+extern s32 D_us_80178B78;
+
+#ifdef VERSION_PSP
+extern u16 D_91F8618[];
+extern u16 D_psp_092F1138[];
+#endif
+
+#define CLUT_INDEX_SERVANT_2 0x1410
+extern u16 g_ServantClut[16];
+extern u16 g_SwordClut[32];
+extern VECTOR g_TransferVector;
+extern SVECTOR g_RotationAngle;
+
+extern VECTOR D_us_80170080;
+
+#ifdef VERSION_PSP
+extern u16 D_91F8618[];
+extern u16 D_psp_092F1138[];
+#endif
+
+void ServantInit(InitializeMode mode) {
+    Entity* self;
+    s32 i;
+    s32 j;
+    u16* dst;
+    u16* src;
+    s32 posX;
+    RECT rect;
+
+    if (mode == MENU_SWITCH_SERVANT || mode == MENU_SAME_SERVANT) {
+        ProcessEvent(NULL, true);
+        if (mode == MENU_SAME_SERVANT) {
+            return;
+        }
+    }
+
+    self = &g_Entities[UNK_ENTITY_4];
+
+    dst = &g_Clut[CLUT_INDEX_SERVANT];
+    src = g_ServantClut;
+
+#ifdef VERSION_PSP
+    for (j = 1; i < LEN(g_SwordClut); j++) {
+        g_SwordClut[j] |= 0x8000;
+    }
+#endif
+
+    for (i = 0; i < LEN(g_ServantClut); i++) {
+        *dst++ = *src++;
+    }
+
+    dst = &g_Clut[CLUT_INDEX_SERVANT_2];
+    src = g_SwordClut;
+
+    for (i = 0; i < LEN(g_SwordClut); i++) {
+        *dst++ = *src++;
+    }
+
+    rect.x = 0;
+    rect.y = 0xF4;
+    rect.w = 0x30;
+    rect.h = 1;
+
+    dst = &g_Clut[CLUT_INDEX_SERVANT];
+    LoadImage(&rect, (u_long*)dst);
+    DestroyEntity(self);
+
+    self->zPriority = PLAYER.zPriority - 2;
+    self->facingLeft = (PLAYER.facingLeft + 1) & 1;
+
+    if (mode == MENU_SWITCH_SERVANT) {
+#ifdef VERSION_PSP
+        if ((D_8003C708.flags & FLAG_UNK_20) ||
+            (D_8003C708.flags & FLAG_UNK_40)) {
+#else
+        if ((LOW(D_8003C708.flags) & (FLAG_UNK_40 | FLAG_UNK_20)) != 0) {
```

Comment:
The PSP version doesn't match on PS1? Seems like an optimization

Reply:
I'll check again.

Reply:
Yeah, I just tried again and couldn't get them to match. The load is different (word vs half) as well. Possibly a bug in the original (loading the word) leading to the logic change. 🤷